Upgrade `release-toolkit`

mokagio opened the pull request:
### Fix
<!--
***(Required)*** Add a concise description of what you fixed.  If this is related to an issue, add a link to it.  If applicable, add screenshots, animations, or videos to help illustrate the fix.
-->

### Test
<!--
***(Required)*** List the steps to test the behavior.  For example:
1. Go to...
2. Tap on...
3. See error...
-->

### Review
<!--
***(Required)*** Add instructions for reviewers.  For example:
Only one developer and one designer are required to review these changes, but anyone can perform the review.
-->

### Release
<!--
***(Required)*** Add a concise statement to `RELEASE-NOTES.txt` if the changes should be included in release notes. Include details about updating the notes in this section. For example:
`RELEASE-NOTES.txt` was updated in d3adb3ef with:
> Added markdown support
-->
<!--
If the changes should not be included in release notes, add a statement to this section. For example:
These changes do not require release notes.
-->

mokagio committed:
Update strings for translation - version 2.33

mokagio committed:
Use named keyword arguments in `build_pre_releases`

mokagio committed:
Disable `android_build_prechecks` because of a bug in old r-t version

r-t = release-toolkit

notandyvee committed:
Go back one screen when confirming a code response says it expired.

mokagio committed:
Update strings for translation - version 2.33 (#1674)

notandyvee committed:
Merge pull request #1672 from Automattic/andy/firebird-issue-1670

Go back one screen when confirming a code response says it expired.

mokagio committed:
Switch to WIP release-toolkit version

mokagio committed:
Tidy: Remove noisy comments and homogenize constants in `Fastfile`

mokagio committed:
Replace `setfrozentag` with `set_milestone_frozen_marker`

mokagio committed:
Extract `versionName` and `versionCode` to `version.properties`

mokagio committed:
Add `Versioning` methods to with `release-toolkit` v10+

mokagio committed:
Use name block arguments instead of `options` in `code_freeze`

mokagio committed:
Convert `release_notes_path` method to constant

mokagio committed:
Remove `android_tag_build` usages

mokagio committed:
Update `code_freeze` to use `Versioning` methods

mokagio committed:
Extract release branch name computation in name

This is handy for DRY but also to change the name during testing

mokagio committed:
DRY `./Simplenote` path in `APP_SOURCES_FOLDER` const

mokagio committed:
DRY metadata folder path in const

mokagio committed:
DRY build folder path

mokagio committed:
Rename `update_appstore_strings` to `update_play_store_strings`

mokagio committed:
Use named block arguments in `update_play_store_strings` lane

mokagio committed:
Make `update_play_store_strings` commit the update, if any

mokagio committed:
Replace unavailable `android_betabuild_prechecks` with individual checks

mokagio committed:
Use `Versioning` in `new_beta_release`

mokagio committed:
Use `Versioning` in `new_hotfix_release`

mokagio committed:
Use `Versioning` in `finalize_hotfix_release`

mokagio committed:
Extract lanes to check translations and DRY GlotPress URLs

mokagio committed:
DRY `res/` path into const

mokagio committed:
Use `Versioning` in `finalize_release`

mokagio committed:
Use named block arguments + refactor `build_and_upload_apk`

mokagio commented on fastlane/Fastfile:
```diff
@@ -9,14 +9,26 @@ USER_ENV_FILE_PATH = File.join(Dir.home, '.simplenoteandroid-env.default')
 PROJECT_ROOT_FOLDER = File.dirname(File.expand_path(__dir__))
-GITHUB_REPO = 'automattic/simplenote-android'
+
+ORGANIZATION = 'automattic'
+REPO_NAME = 'simplenote-android'
+GITHUB_REPO = "#{ORGANIZATION}/#{REPO_NAME}".freeze
 DEFAULT_BRANCH = 'trunk'
+BUILDKITE_ORG = ORGANIZATION
+BUILDKITE_PIPELINE = REPO_NAME
```
The idea with these over-engineered constants is that they should be easy to copy paste into other `Fastfile`s.

The more homogeneous we can keep the `Fastfile`s the easier it should be to maintain them.

mokagio commented on fastlane/Fastfile:
```diff
@@ -37,66 +49,79 @@ SUPPORTED_LOCALES = [
   ENV['validate_translations'] = 'buildRelease'
   ENV['FL_RELEASE_TOOLKIT_DEFAULT_BRANCH'] = 'trunk'
 
-  ########################################################################
-  # Release Lanes
-  ########################################################################
-  #####################################################################################
-  # code_freeze
-  # -----------------------------------------------------------------------------------
-  # This lane executes the steps planned on code freeze
-  # -----------------------------------------------------------------------------------
-  # Usage:
-  # bundle exec fastlane code_freeze [update_release_branch_version:<update flag>] [skip_confirm:<skip confirm>]
-  #
-  # Example:
-  # bundle exec fastlane code_freeze
-  # bundle exec fastlane code_freeze update_release_branch_version:false
-  # bundle exec fastlane code_freeze skip_confirm:true
-  #####################################################################################
```
I decided to remove all these comments. Some were out of date, all felt to me like noise in the source.

mokagio commented on fastlane/Fastfile:
```diff
@@ -37,66 +49,79 @@ SUPPORTED_LOCALES = [
-  # bundle exec fastlane code_freeze update_release_branch_version:false
-  # bundle exec fastlane code_freeze skip_confirm:true
-  #####################################################################################
   desc 'Creates a new release branch from the current trunk'
```
```suggestion
  desc 'Creates a new release branch from the current default branch'
```

mokagio committed:
Replace `trunk` with "default branch" in a `desc` string

mokagio commented on fastlane/Fastfile:
```diff
@@ -708,3 +684,61 @@ platform :android do
+  ensure_git_branch(branch: "^#{RELEASE_BRANCH_ROOT}")
+end
+
+RELEASE_BRANCH_ROOT = 'release/'
```
This const _could_ go at the top with all the others, but since it's used only by the two release branch methods, it felt clearer to put it at the bottom here, after its consumers.

It's a bit inconsistent but I think it works.

Happy to move it up with the rest if you think it's not worth breaking the pattern.

AliSoftware committed:
Update Ruby dependencies

AliSoftware commented:
FYI I've just opened https://github.com/Automattic/simplenote-android/pull/1677 to fix CVE on `rexml`… which might conflict with your draft PR here

AliSoftware committed:
Merge pull request #1677 from Automattic/tooling/update-ruby-dependencies

[Tooling] Update Ruby Dependencies

AliSoftware commented:
Note that while #1677 fixed the `rexml`-related CVEs, there's currently [one remaining CVE about `nokogiri` still open](https://github.com/Automattic/simplenote-android/security/dependabot/73).

Fixing it will require to update `release-toolkit` to a newer version, as `nokogiri` is a dependency of it and that's where the version constraint on it comes from. Since the update of `release-toolkit` is already planned to be handled by this very PR, I'm leaving it to @mokagio and this PR to fix that last CVE as a side-effect of updating `release-toolkit`.

mokagio committed:
Merge remote-tracking branch 'origin/trunk' into mokagio/release-toolkit-v12

Merge conflicts on `Gemfile.lock`. Resolved by:

1. `git checkout --theirs Gemfile.lock` to get all the changes from
   30a35ecc2d8cce1380146a3d8d24ee2f722c0f33 (#1677)
2. `bundle install` to switch `release-toolkit` back to the WIP branch
   we're working on here.

mokagio committed:
Move trailing `/` in GlotPress URL to avoid redirect

mokagio committed:
Rename `*_PLAYSTORE_*` to `*_STORE_*` to make generic

This will allow copy-pasting the constants in other projects for a more
homogeneous setup.

mokagio committed:
Add missing trailing `/` to release notes GlotPress URL

mokagio commented:
Closed in favor of #1680
